src/ruby/yast/core_ext.rb

```diff
@@ -0,0 +1 @@
+require_relative "core_ext/string"
```

Comment:
problem is if you require extension in any file, then it is in all files everywhere (if you mean clean string extension not refinement). Any of problem with this extending class is that it can also cause collision. Consider that you use in an yast module another core extension, that have same method, just with slightly different implementation and then it is only on execution order which one is used and can cause very nasty effects.
BTW another approach, I like maybe even more and is more object oriented is to create
```
AnsiString < ::String
  def remove_ansi_sequences
  ...
  ...
end
```
so it is clear, that you have instance of string with ansi characters and it have own methods.
About refinements, I think it is quite good solution, just it quite new, so I don't yet have much experience about it, so do not know yet weak sides.